utils/navigator_robotx_comms/nodes/navigator_robotx_comms.py

```diff
@@ -0,0 +1,157 @@
+                                                                                                          delim,
+                                                                                                          time_string,
+                                                                                                          delim,
+                                                                                                          latitude,
+                                                                                                          delim,
+                                                                                                          north_south,
+                                                                                                          delim,
+                                                                                                          longitude,
+                                                                                                          delim,
+                                                                                                          east_west,
+                                                                                                          delim,
+                                                                                                          self.team_id,
+                                                                                                          delim, str(
+                self.system_mode), delim, str(self.auv_status), checksum)
+
+        return msg_return
+
+    def auv_status_callback(self, auv_status):
+
+        self.auv_status = auv_status
+
+    def gps_coord_callback(self, lla):
+
+        self.gps_array = lla
+
+    def gps_odom_callback(self, odom):
+
+        self.odom = odom
+
+
+if __name__ == "__main__":
+
+    # INIT TCP
+    TCP_IP = rospy.get_param('td_ip')
+    TCP_PORT = rospy.get_param('td_port')
+    BUFFER_SIZE = 1024
+    connected = False
+
+    socketConnection = None
+
+    robot_x_comms = RobotXComms()
+    rospy.Subscriber("lla", PointStamped,  robot_x_comms.gps_coord_callback)
+    rospy.Subscriber("odom", Odometry, robot_x_comms.gps_odom_callback)
+
+    while not rospy.is_shutdown():
+        if not connected:
+            print 'Attempting Connection to TD Server'
+        while not connected:
+            connected = False
+            # recreate socket
+            socketConnection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            # attempt to reconnect, otherwise sleep for 2 seconds
+            try:
+                socketConnection.connect((TCP_IP, TCP_PORT))
+                connected = True
+                print 'Connection to TD Server Successful'
+            except socket.error:
+                time.sleep(2)
+        heartbeatMsg = robot_x_comms.heartbeat_msg()
+        while True:
```

Comment:
We avoid ```while True``` as it can sometimes make your program impossible to close. Consider using ```while not rospy.is_shutdown()```, or even better using rospy.Timer to have ROS handle this regular interval function call for you